__tests__/app.test.js

```diff
@@ -34,7 +35,43 @@ describe('GET /api/topics', () => {
             response.body.topics.forEach((topic) => {
             expect(typeof topic.description).toBe('string');
             expect(typeof topic.slug).toBe('string');
+            expect.objectContaining({
+                description: 'The man, the Mitch, the legend',
+                slug: 'mitch'
+              })
         });
       });
     });
+})
+
+describe('GET /api', () => {
+    test('GET: 200 send an object describing all the available endpoints on your API', async () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((response) => {
+            let existingPaths = []
+            let pathsInJsonFile = []
+            let result = true;
+
+            app._router.stack.forEach(function(r){
+                if (r.route && r.route.path){
+                    existingPaths.push(r.route.path)
+                }
+              })
+
+            for (const key in response.body) {
+                const splittedKey = key.split(' ')
+                pathsInJsonFile.push(splittedKey[1])
+            }
+
+            for(let element of existingPaths) {
+                if (!pathsInJsonFile.includes(element)) {
+                   result = false;
+                }
+             }
+
+            expect(result).toBe(true)
+        })
+    })
```

Comment:
I can see that you've put a lot of work into making this work, however it's not very human readable. We know what our endpoints should look like as we have access to the endpoints.json file. If you parse the returned body you should be able to compare the two.